tempodb/encoding/vparquet3/nested_set_model.go

```diff
@@ -97,6 +99,7 @@ func assignNestedSetModelBounds(trace *Trace) bool {
 	for _, root := range rootNodes {
 		node := root
 		node.span.NestedSetLeft = nestedSetBound
+		node.span.ParentID = nestedSetRootParent
```

Comment:
1) This won't be present on existing data, does that matter?

2) Can you explain a bit which use case needs it?  Maybe NotSibling and NotAncestor? Sibling should be ok because either 0 legit or 0 unset, root nodes don't have siblings.

Reply:
> This won't be present on existing data, does that matter?

I think it's fine. We've marked these as experimental and it only impacts `{} !~ {}` on root spans.

NotSibling is the only one that matters. This allows it to distinguish between a root span and a span that is disconnected in the trace. Ancestor/Descendant only checks left/right so this doesn't impact that algorithm.

Reply:
Oh sorry I meant NotParent (child, invert, falseforall).

Reply:
Agree if it's a graceful fallback that sounds good.  To clarify it means that `{ } !~ { }` will skip existing roots (parent=0), and going forward it will pick them up (parent=-1).   Can you add a comment?

Reply:
So this?
```
{} !< {}
```
I think this worked before b/c child/parent pulls both NestedSetLeft and NestedSetParent and NestedSetLeft on Root was nonzero. Therefore it passed filters and made it to the engine.

The reason
```
{} !~ {}
```
did not work is b/c this only pulls nested set parent and [these lines](https://github.com/grafana/tempo/blob/419288d49dae887d8887aae24e83f3407665f8f1/tempodb/encoding/vparquet3/block_traceql.go#L1666-L1672) dropped the span. The code was basically seeing the span as having no valid attributes, so it didn't match anything and so just let 'er go.

Reply:
> Agree if it's a graceful fallback that sounds good. To clarify it means that { } !~ { } will skip existing roots (parent=0), and going forward it will pick them up (parent=-1). Can you add a comment?

Sorry, this comment wasn't visible when I replied. The real fix was more like:

If you requested NestedSetParent ONLY from the fetch layer it would fail to return root spans due to the filtering linked above. It drops root spans even though we have actual information about the parent of a root span. So by using -1 we are giving a clear signal to the fetch layer that we have valid information about the parent of this span b/c normally 0 means "we don't know anything".

Will add a comment.